cpp/scenario/gazebo/CMakeLists.txt

```diff
@@ -94,6 +94,11 @@ target_link_libraries(ScenarioGazebo
 set_target_properties(ScenarioGazebo PROPERTIES
     PUBLIC_HEADER "${SCENARIO_PUBLIC_HDRS}")
 
+if(NOT CMAKE_BUILD_TYPE STREQUAL "PyPI")
```

Comment:
This will make life complicated for anyone packaging the project in any form, being it as part of a superbuild, as Debian packages or a conda recipe. Instead of conflating build logic and packaging logic, could we have an explicit CMake variable to select if the `gym-ignition`  project get installed in User mode or Developer mode, and then just specify the default based on the `CMAKE_BUILD_TYPE` , something  like: 
~~~cmake
if(NOT CMAKE_BUILD_TYPE STREQUAL "PyPI")
  set(GYM_IGNITION_DEVELOPER_MODE_DEFAULT ON)
else()
   set(GYM_IGNITION_DEVELOPER_MODE_DEFAULT OFF)
endif() 
option(GYM_IGNITION_DEVELOPER_MODE "If ON, gym-ignition is installed in Developer mode, otherwise in user mode" ${GYM_IGNITION_DEVELOPER_MODE_DEFAULT})
if(GYM_IGNITION_DEVELOPER_MODE)
    target_compile_options(ScenarioGazebo PRIVATE
        -DGYMIGNITION_CMAKE_INSTALL_PREFIX="${CMAKE_INSTALL_PREFIX}")
endif()
~~~

Reply:
Thanks for the suggestion of propagating the build type as normal option. I will have a look at it. I don't remember precisely why we had to define a new build type for PyPI in the first place, for sure I tried to use a normal CMake variable as first attempt and then I faced problems that led me to this build type workaround, that is much more convoluted. What I'm sure is that I didn't do it if there were more simple solutions :)

I am planning to give a refreshing also to the CMake project, I will keep you comment in mind.

> This will make life complicated for anyone packaging the project in any form, being it as part of a superbuild, as Debian packages or a conda recipe.

This is something I don't understand. The PyPI type is not meant to be used by any downstream user or packager. It is meant to be used to create packages for PyPI, as the name. Any other usage are not supported.

Reply:
> > This will make life complicated for anyone packaging the project in any form, being it as part of a superbuild, as Debian packages or a conda recipe.
> 
> This is something I don't understand. The PyPI type is not meant to be used by any downstream user or packager. It is meant to be used to create packages for PyPI, as the name (and we are the maintainers of the packages). Any other usage is not supported.

Exactly. With this modification, now the only way to properly install `gym-ignition` in a relocatable way is to set the `CMAKE_BUILD_TYPE` to `PyPI`, and I guess that this is not supported if you are not installing for `PyPI`. I guess we want to support "proper" installation also if the project is not installed with `PyPI` .

Reply:
Ow ok now I understood your initial point: supporting a relocatable installation which could be implemented in the same way we package the project for PyPI.

Note the using PyPI for this purpose is wrong. In fact, due to how the shared libraries must be placed in the `site_package` folder, you need to have the following installation tree:

```
install_tree/
├── gym_ignition
│   ├── [...]
│   ├── __init__.py
│   └── plugins
│       ├── libControllerRunner.so
│       ├── libECMProvider.so
│       ├── libJointController.so
│       └── libPhysicsSystem.so
├── gym_ignition_environments
│   ├── __init__.py
│   └── [...]
├── libECMSingleton.so
├── scenario_bindings.py
└── _scenario_bindings.so
```

As you can see, shared libraries are installed in the root of the `CMAKE_INSTALL_PREFIX`, and plugins inside `gym_ignition/plugins`. This is not what you want to obtain in a regular installation! In light of this, we should start thinking how to make coexist a (static) relocatable Developer installation with the PyPI build type without messing too much with the CMake project.

Reply:
>  This is not what you want to obtain in a regular installation! In light of this, we should start thinking how to make coexist a (static) relocatable Developer installation with the PyPI build type without messing too much with the CMake project.

Yes, we can also have a separate issue for that.